cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -594,7 +594,15 @@ Status NumPyConverter::Visit(const FixedSizeBinaryType& type) {
 
   if (mask_ != nullptr) {
     Ndarray1DIndexer<uint8_t> mask_values(mask_);
-    RETURN_NOT_OK(builder.AppendValues(data, length_, mask_values.data()));
+    RETURN_NOT_OK(builder.Reserve(length_));
+    for (int64_t i = 0; i < length_; ++i) {
+      if (mask_values[i]) {
+        RETURN_NOT_OK(builder.AppendNull());
+      } else {
+        RETURN_NOT_OK(builder.Append(data));
+      }
+      data += stride_;
```

Comment:
Does this solve the strided conversion case? If so, perhaps you can add a test for it?

Reply:
Sadly not, I expected it would, but I wrote some tests and it wasn't sufficient. That's why I made https://issues.apache.org/jira/browse/ARROW-12667 as a follow up issue. So that I can test it for all various types and make sure it works in all cases.

Reply:
Also added tests and fix for strided binary arrays (with and without mask)